### DISCUSSION TITLE
Safe Lines with Drag-and-Drop Node Reference Code in GDScript

### DISCUSSION THREAD

**Discussion by Dark-Horizons** [godotengine/godot-proposals#11458] [Scripting]
Currently, when dragging a node from the scene tree to a script, the generated code uses the following format:
`@onready var sound_fx: AudioStreamPlayer = $SoundFX`

While this works, it leads to "unsafe line number" warnings. To fix this issue, I manually adjust it to:
`@onready var sound_fx := $SoundFX as AudioStreamPlayer`

This format avoids the warning and ensures type safety.
However, performing this adjustment manually every time can be tedious.

Is there a solution for this? If not, could it be added as an option in the editor settings?

**Comment by dalexeev** [comment]
* See https://github.com/godotengine/godot-proposals/issues/9966

> Safe lines do not always mean better or more reliable code. See the note above about the `as` keyword. For example:
>
> ```gdscript
> @onready var node_1 := $Node1 as Type1 # Safe line.
> @onready var node_2: Type2 = $Node2 # Unsafe line.
> ```
>
> Even though `node_2` declaration is marked as an unsafe line, it is more reliable than `node_1` declaration. Because if you change the node type in the scene and accidentally forget to change it in the script, the error will be detected immediately when the scene is loaded. Unlike `node_1`, which will be silently cast to `null` and the error will be detected later.
